src/payment/payment.ts

```diff
@@ -31,6 +32,10 @@ export interface PaymentInstrumentMeta {
     deviceSessionId?: string;
 }
 
+export interface AccountCreationInstrument {
+    shouldCreateAccount?: boolean;
```

Comment:
Since in the typeguard you check for it not being undefined, does it need to be undefined here?

Reply:
I will fix the typeguard